### DISCUSSION TITLE
When switching relation manager tabs, getStateUsing fields reset to 0

### DISCUSSION THREAD

**Discussion by joshembling** [filamentphp/filament#6790] [General]
In one of my Relation Managers, I have a TextColumn that is not a field in the database, but a small calculation.

See below:
```
Tables\Columns\TextColumn::make('occurrences')
                    ->getStateUsing(function (Model $record) {
                        return $record->logs
                            ->where('id', request()->route()->record)
                            ->where('error_id', $record->id)
                            ->count();
                    }),
```

On page load this will correctly show as the model count. But it seems when switching tabs, Livewire? is ignoring this and just checking for db fields.

Can someone point me in the right direction in correcting this?

**Comment by danharrin** [comment]
You're just accessing the record wrong, should be

```
->getStateUsing(function (Model $record, $livewire) {
    return $record->logs
        ->where('id', $livewire->ownerRecord->id)
        ->where('error_id', $record->id)
        ->count();
})
```

However, its best to move this to a query subselect, because this is currently an N+1 problem
